packages/compass-indexes/src/modules/create-index/index.js

```diff
@@ -165,14 +160,13 @@ export const createIndex = () => {
     });
 
     const options = {};
-    options.background = state.isBackground;
     options.unique = state.isUnique;
     // The server will generate a name when we don't provide one.
     if (state.name !== '') {
       options.name = state.name;
     }
     if (state.isCustomCollation) {
-      options.collation = state.collation;
+      options.collation = state.collationString.value;
```

Comment:
I'm not sure about storing the `.value` in the state, is not really used as a react state, you can just `parse(state.collationString.text)` and avoid introducing statefulness here.

To me the same goes for `isValid`: we can simplify everything and store only the string, parse the option here and handle the parsing exception nicely.

We stopped doing this kind of on-the-fly validations everywhere. It simplifies the code and having them is not really helpful cause it would trigger multiple times while the user is typing and give a negative signal while there is nothing actually bad going on.

Reply:
Same as above.

Reply:
Actually thinking it twice I'm not too sure, cause really none of the state that was here seems a fit for redux (we probably don't want to change it right away?). Is all of a volatile form state that doesn't seem very global to me. Regardless, probably none of what I'm saying makes sense, @gribnoysup recently reviewed a ton of best practices etc, maybe he can give us a pointer.

Reply:
> We stopped doing this kind of on-the-fly validations everywhere

Is it something we agreed on and have documented somewhere? This seems like an important thing to be aware of, but it's the first time I hear about this 😅 Should we add this to "component conventions" or something like that? Also I wonder how this aligns with what we heard on the last design sync where people were asking for syntax errors highlights right when they are typing in code editor inputs, is it something that should apply only to very explicit forms like "create / edit" modals and connection form?

> Is all of a volatile form state that doesn't seem very global to me

That would always depend on the use case, but yes, Redux docs mention that form state [should not be part of the store](https://redux.js.org/style-guide/#avoid-putting-form-state-in-redux). FWIW this only applies to very straightforward cases where you have an actual form that is totally isolated from everything else and which state can be completely discarded on submit or close / cancel. Which does match this modal case, but would require a bunch of additional refactoring: moving only one value to the component state and keeping all the other input state still in the store doesn't make much sense to me and will create a bigger mess from my perspective.

I do think though that we can avoid producing the parsed value on every edit and keeping it in the state and instead only keep the collation string in the redux store. I see that this was modeled following the aggregation plugin code, but the compass-aggregation case is way different: we do show validation error there right away when you type and so we have to parse / validate on edit _and_ this state is used in multiple different actions across the plugin code and so it needs to be "globally" available. I see that here you don't use the validation information at all and the parsed value is only used when form is submitted, so that's probably the only place where you would want to parse / validate the value. Also other form values here seem to [dispatch a `handleError` action](https://github.com/mongodb-js/compass/blob/a364e83f811047ede797cdee5dc3d9e29cce1ac4/packages/compass-indexes/src/modules/create-index/index.js#L156) when validation fails, it's probably something you would want to do in this action for collation too.